hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/HoodieDeltaStreamer.java

```diff
@@ -112,9 +112,14 @@ public HoodieDeltaStreamer(Config cfg, JavaSparkContext jssc, FileSystem fs, Con
   public HoodieDeltaStreamer(Config cfg, JavaSparkContext jssc, FileSystem fs, Configuration conf,
                              TypedProperties props) throws IOException {
     // Resolving the properties first in a consistent way
-    this.properties = props != null ? props : UtilHelpers.readConfig(
-        FSUtils.getFs(cfg.propsFilePath, jssc.hadoopConfiguration()),
-        new Path(cfg.propsFilePath), cfg.configs).getConfig();
+    if (props != null) {
+      this.properties = props;
+    } else {
```

Comment:
can this be written more readable? using if-elseif blocks?

Reply:
sure

Reply:
Done.